service/dynamodb/dynamodbattribute/field.go

```diff
@@ -0,0 +1,267 @@
+package dynamodbattribute
+
+import (
+	"reflect"
+	"sort"
+	"strings"
+)
+
+type field struct {
+	tag
+
+	Name        string
+	NameFromTag bool
+
+	Index []int
+	Type  reflect.Type
+}
+
+func fieldByName(fields []field, name string) (field, bool) {
+	for _, f := range fields {
```

Comment:
Im guessing this is another optimization, but compressing these for loops into one is also possible.